pkg/discovery/dtofactory/general_builder.go

```diff
@@ -381,16 +385,63 @@ func aggregateContainerThrottlingSamples(entityID string, samples []metrics.Thro
 	sort.SliceStable(samples, func(i, j int) bool {
 		return samples[i].Timestamp < samples[j].Timestamp
 	})
-	throttled = (samples[numberOfSamples-1].Throttled - samples[0].Throttled)
-	total = samples[numberOfSamples-1].Total - samples[0].Total
 
+	resetIndexes := []int{}
```

Comment:
Just to confirm. If I understand correctly, a map is not needed here because counter reset is so rare, and even if there is a reset, it would be extremely unlikely to have multiple resets between two discoveries, right?

Reply:
I didn't give it much thought wrt performance while implementing this, but your arguments are all valid. Additionally this slice would atmost be size 10 max (unless the sample collection per discovery is configured to something outrageously low), where the additional memory and  the lookup overhead of a map might result in similar (if not worse) performance.